op-bindings/bindgen/etherscan-contracts.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type etherscanContract struct {
+	Name             string
+	DeployedAddress  string
+	PredeployAddress string
+	Abi              string
+	Bytecode         string
+}
+
+type etherscanApiResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+	Result  string `json:"result"`
+}
+
+type etherscanContractMetadata struct {
+	Name              string
+	DeployedBin       string
+	Package           string
+}
+
+func readEtherscanContractsList(filePath string) ([]etherscanContract, error) {
+	var contracts []etherscanContract
+	err := readJSONFile(filePath, &contracts)
+	return contracts, err
+}
+
+func fetchEtherscanData(url string) (string, error) {
+	// TODO Handle rate limit error
+	resp, err := http.Get(url)
```

Comment:
There is a retry helper in `op-service` someplace

Reply:
Decided to add my own simple retry logic since handling of the error is just a matter of waiting 2 seconds and trying again